src/tree/context-dep.h

```diff
@@ -20,6 +20,7 @@
 #ifndef KALDI_TREE_CONTEXT_DEP_H_
 #define KALDI_TREE_CONTEXT_DEP_H_
 
+#include <set>
```

Comment:
No need for this include, as I think you do not use 'set'.  If you want, you can include util/stl-utils.h, which includes unordered_set (with suitable ifdefs).